Public/Src/Cache/DistributedCache.Host/Service/DistributedCacheServiceFacade.cs

```diff
@@ -68,5 +71,17 @@ public static async Task RunAsync(DistributedCacheServiceArguments arguments)
                 }
             }
         }
+
+        private static async Task<BoolResult> ShutdownWithTimeout(Context context, LocalContentServer server, TimeSpan timeout)
+        {
+            var shutdownTask = server.ShutdownAsync(context);
+            if (await Task.WhenAny(shutdownTask, Task.Delay(timeout)) != shutdownTask)

```

Comment:
No cancellation or result processing of Task that lost?

Reply:
Task.Delay can't throw and if `shutdownTask` is not yet finished, then the process will exit anyways. So in this case, it is perfectly fine. I wanted to use our extension method `WithTimeout` but unfortunately this assembly doens't reference Bxl.Utilities and we don't wat to add this reference.